src/ray/rpc/raylet/raylet_client.cc

```diff
@@ -131,11 +126,13 @@ ray::Status RayletClient::GetTask(std::unique_ptr<ray::TaskSpecification> *task_
 }
 
 ray::Status RayletClient::TaskDone() {
+  RETURN_IF_DISCONNECTED(is_connected_);
   TaskDoneRequest task_done_request;
   task_done_request.set_worker_id(worker_id_.Binary());
 
-  auto callback = [](const Status &status, const TaskDoneReply &reply) {
-    if (!status.ok()) {
+  auto callback = [this](const Status &status, const TaskDoneReply &reply) {
```

Comment:
One option to help make sure we're consistent with the reply callbacks is to add a templated helper function to return the callback, something like this:
```c++
template <typename ReplyT>
std::function<void(const &Status, const &ReplyT)> MakeReplyCallback() {
  auto callback = ...
  return callback;
}
```

Reply:
It seems not necessary, because we still need to add the actual content of callback.